engine/games/voltorb_flip.asm

```diff
@@ -0,0 +1,1617 @@
+	ld a, [hli]
+	and VOLTORB_FLIP_CARD_NUMBER_MASK | VOLTORB_FLIP_CARD_FLIPPED
+	cp 2
+	jr nz, .skip2add
+	inc b
+.skip2add
+	cp 3
+	jr nz, .skip3add
+	inc b
+.skip3add
+	dec c
+	jr nz, .check23left
+	ld a, b
+	ld [wVoltorbFlipUnflippedMultipliers], a ; for the are you sure song
+	and a
+	ret nz
+	call VFRefreshMap
+	ld hl, wVoltorbFlipCursorX
+	ld a, [hli]
+	ld b, a
+	ld a, [hli]
+	ld c, a
+	di
+	call VFRefreshScreen
+	ei
+	ld de, SFX_3RD_PLACE ; level clear
+	call WaitPlaySFX
+	ld c, VOLTORB_FLIP_LEVEL_CLEAR_DELAY_FRAMES
+	call DelayFrames
+	ld a, [wVoltorbFlipLevel]
+	inc a
+	cp VOLTORB_FLIP_LEVEL_CAP_GUARD
+	call z, .caplevel
+	ld [wVoltorbFlipLevel], a
+	ld de, SFX_TRANSACTION
+	call WaitPlaySFX
+	ld hl, wVoltorbFlipCoinsThisGame
+	ld a, [hli]
+	ld d, a
+	ld a, [hld]
+	ld e, a
+	push hl
+	ld hl, wCoins
+	ld a, [hli]
+	ld l, [hl]
+	ld h, a
+	add hl, de
+	push hl
+	pop de
+	pop hl
+	ld a, d
+	cp HIGH(MAX_COINS)
+	jr nc, .maybecap
+.totalcoins
+	ld a, d
+	ld [wCoins], a
+	ld a, e
+	ld [wCoins + 1], a
+	jp VFInitLevel
+.caplevel
+	ld a, VOLTORB_FLIP_LEVEL_LIMIT
+	ret
+.maybecap
+	cp HIGH(MAX_COINS)
+	jr nz, .capcoins
+	ld a, e
+	cp LOW(MAX_COINS)
+	jr c, .totalcoins
+; fallthrough
+.capcoins
+	ld de, MAX_COINS
+	jr .totalcoins
+
+.bbutton
+	and VOLTORB_FLIP_DIRECTION_MASK
+	ret z
+	ld b, a
+	ldh a, [hJoypadPressed]
+	ld c, a
+	ld a, [de]
+	bit B_PAD_DOWN, b
+	call nz, .bdown
+	bit B_PAD_UP, b
+	call nz, .bup
+	bit B_PAD_LEFT, b
+	jr nz, .bleft
+	bit B_PAD_RIGHT, b
+	ret z
+.bright
+	bit B_PAD_DOWN, c
+	jr nz, .toggle3
+	bit B_PAD_UP, c
+	jr nz, .toggle1
+	ret
+.bleft
+	bit B_PAD_DOWN, c
+	jr nz, .toggle2
+	bit B_PAD_UP, c
+	jr nz, .toggle0
+	ret
+.bup
+	bit B_PAD_LEFT, c
+	jr nz, .toggle0
+	bit B_PAD_RIGHT, c
+	jr nz, .toggle1
+	ret
+.bdown
+	bit B_PAD_LEFT, c
+	jr nz, .toggle2
+	bit B_PAD_RIGHT, c
+	jr nz, .toggle3
+	ret
+.toggle0
+	xor VOLTORB_FLIP_CARD_MARKED0
+	ld [de], a
+	ret
+.toggle1
+	xor VOLTORB_FLIP_CARD_MARKED1
+	ld [de], a
+	ret
+.toggle2
+	xor VOLTORB_FLIP_CARD_MARKED2
+	ld [de], a
+	ret
+.toggle3
+	xor VOLTORB_FLIP_CARD_MARKED3
+	ld [de], a
+	ret
+
+.rightbutton
+	ld a, b
+	inc a
+	cp VOLTORB_FLIP_BOARD_SIZE
+	jr nz, .norowwrap
+	xor a
+.norowwrap
+	ld [wVoltorbFlipCursorX], a
+	ret
+.leftbutton
+	ld a, b
+	dec a
+	cp $ff
+	jr nz, .norowwrap
+	ld a, VOLTORB_FLIP_MAX_INDEX
+	jr .norowwrap
+
+.upbutton
+	ld a, c
+	dec a
+	cp $ff
+	jr nz, .nocolwrap
+	ld a, VOLTORB_FLIP_MAX_INDEX
+.nocolwrap
+	ld [wVoltorbFlipCursorY], a
+	ret
+.downbutton
+	ld a, c
+	inc a
+	cp VOLTORB_FLIP_BOARD_SIZE
+	jr nz, .nocolwrap
+	xor a
+	jr .nocolwrap
+
+VFLoadGFX:
+	ld hl, rSTAT
+	ld a, 1
+.wait
+	ld a, [hl]
+	and STAT_MODE
+	cp STAT_VBLANK
+	jr nz, .wait
+	dec hl ; rLCDC
+	res B_LCDC_ENABLE, [hl]
+	lb bc, BANK(VoltorbFlipGFX), 128 ; load the full gfx sheet so number tiles are present
+	ld de, VoltorbFlipGFX
+	ld hl, vTiles2
+	jp Copy2bpp
+
+VFLoadPalette:
+	ldh a, [rSVBK] ; $ff00+$70
+	push af
+	ld a, BANK(wBGPals2)
+	ldh [rSVBK], a
+	lb bc, $e4, 8
```

Comment:
Here `b` is a value like the ones for `rOBP0`, so it should be in binary: `%11_10_01_00`.